### ISSUE TITLE
ExecutionTest::DerivativesTest is failing on recent Windows builds

### ISSUE THREAD

**Issue by hekota** [microsoft/DirectXShaderCompiler#6180]
**Description**
`ExecutionTest::DerivativesTest` is failing on recent Windows builds. Reproes on x64 and arm64.

**Steps to Reproduce**
On pre-release Windows build 23H2 version 26016.1001:
`hcttest exec-filter ExecutionTest::DerivativesTest`

**Actual Behavior**
```
******************** TEST 'clang-taef-exec :: ExecHLSLTests.dll/ExecutionTest::DerivativesTest' FAILED ********************
Test Authoring and Execution Framework v10.68 for arm64

Warning: TAEF will not listen for crashes from Windows Error Reporting because it failed to set up the listener. ([HRESULT 0x80070005] Failed to enable the Microsoft.Windows.FaultReporting TraceLogging provider in an ETW trace.)

Agility SDK not enabled.

Experimental mode enabled.

Debug layer enabled.



StartGroup: ExecutionTest::DerivativesTest

Using Adapter:Microsoft Basic Render Driver

Verifying derivatives in compute shader results

center  ddx_fine: 256.000000, ddy_fine: 224.000000, ddx_coarse: 32.000000, ddy_coarse: 224.000000

.....

Verifying derivatives in mesh shader results

center  ddx_fine: 2.000000, ddy_fine: 1.875000, ddx_coarse: 0.125000, ddy_coarse: 1.875000

Error: Verify: IsTrue(left || CompareFloatULP(CenterDDXFine, 256.0f, ulpTolerance)) [File: C:\__w\1\s\DXC\tools\clang\unittests\HLSLExec\ExecutionTest.cpp, Function: VerifyDerivResults, Line: 3818]

EndGroup: ExecutionTest::DerivativesTest [Failed]
```

**Environment**
- DXC version: `main` as of 1/22/2024
- Pre-release Windows build 23H2 version 26016.1001

**Comment by pow2clk** [comment]
My inclination here as well is a testing bug due to finicky pixel indexing.

**Comment by hekota** [comment]
Updating title and description - this is not ARM64-specific.

**Comment by hekota** [comment]
Fixed by #6311
